lib/AST/ASTContext.cpp

```diff
@@ -5658,7 +5644,8 @@ GenericEnvironment *GenericEnvironment::forOpaqueType(
 
   auto &ctx = opaque->getASTContext();
 
-  auto properties = getOpaqueTypeArchetypeProperties(subs);
+  auto properties = ArchetypeType::archetypeProperties(
+      RecursiveTypeProperties::HasOpaqueArchetype, { }, Type(), subs);
```

Comment:
`properties` is only used to decide the arena below so you can just do `subs->getRecursiveProperties()`

Reply:
Right, they're effectively the same. Using `archetypeProperties` felt more direct, but I can switch over to `subs->getRecursiveProperties()` if you prefer